Sources/SignalProducer.swift

```diff
@@ -1877,6 +1877,58 @@ extension SignalProducerProtocol {
 			}
 		}
 	}
+	
+	/// Creates a new `SignalProducer` that will bucket each received value into
+	/// a group based on the key returned from `grouping`. 
+	///
+	/// Termination events on the original signal are also forwarded to each
+	/// producer group.
+	///
+	/// - parameters:
+	///   - grouping: a closure that determines the grouping key for a given value
+	///
+	/// - returns: A producer of producers amits one producer for each group and forwards
+	///            each value from the original producer to the inner producer corresponding 
+	///            to the group to which the value belongs to (as determined by the key)
+	public func groupBy<Key: Hashable>(_ grouping: @escaping (Value) -> Key) -> SignalProducer<(Key, SignalProducer<Value, Error>), Error> {
+		return SignalProducer<(Key, SignalProducer<Value, Error>), Error> { observer, disposable in
+			let groups = Atomic<[Key: Signal<Value, Error>.Observer]>([:])
```

Comment:
It doesn't seem necessary to specialise it. Is it a workaround to the compiler's weirdness or something?